app/models/concerns/planning_application_status.rb

```diff
@@ -80,6 +80,8 @@ module PlanningApplicationStatus
 
       event :request_correction do
         transitions from: :awaiting_determination, to: :awaiting_correction
+
+        after { audit_created!(activity_type: "challenged", audit_comment: recommendations&.last&.reviewer_comment) }
```

Comment:
Wondering if there is a way to actually pass in the relevant db record because I've just realised that maybe grabbing the last record like this may be subject to a race condition.

Have a look at https://github.com/aasm/aasm/issues/506#issuecomment-532434609

I realise I may not have done this for some previous work but we should probably use this Proc style approach to ensure we are using the right record.

what are your thoughts?